weave/flow/eval2.py

```diff
@@ -0,0 +1,200 @@
+from collections.abc import Iterator
```

Comment:
general nit: "Beta" seems unnecessary? Why include it? Is there a better world that could be more descriptive?